rules/order/src/Rector/Class_/OrderPropertyByComplexityRector.php

```diff
@@ -100,50 +102,53 @@ public function refactor(Node $node): ?Node
     {
         $propertyByVisibilityByPosition = $this->resolvePropertyByVisibilityByPosition($node);
 
-        foreach ($propertyByVisibilityByPosition as $propertyByPosition) {
+        foreach ($propertyByVisibilityByPosition as $resolvePropertyByPosition) {
             $propertyNameToRank = [];
             $propertyPositionByName = [];
 
-            foreach ($propertyByPosition as $position => $property) {
+            foreach ($resolvePropertyByPosition as $position => $property) {
                 /** @var string $propertyName */
                 $propertyName = $this->getName($property);
 
                 $propertyPositionByName[$position] = $propertyName;
-                $propertyNameToRank[$propertyName] = $this->propertyRanker->rank($property);
+                $propertyNameToRank[$propertyName]['rank'] = $this->propertyRanker->rank($property);
+                $propertyNameToRank[$propertyName]['position'] = $position;
             }
 
-            asort($propertyNameToRank);
-            $sortedPropertyByRank = array_keys($propertyNameToRank);
+            $sortedPropertyByRank = $this->getSortedPropertiesByRankAndPosition($propertyNameToRank);
 
-            $oldToNewKeys = $this->stmtOrder->createOldToNewKeys($propertyPositionByName, $sortedPropertyByRank);
+            $oldToNewKeys = $this->stmtOrder->createOldToNewKeys($sortedPropertyByRank, $propertyPositionByName);
 
             $this->stmtOrder->reorderClassStmtsByOldToNewKeys($node, $oldToNewKeys);
         }
 
         return $node;
     }
 
-    private function getVisibilityAsString(Property $property): string
+    /**
+     * @param Property|ClassConst $stmt
+     */
+    private function getVisibilityAsString(Stmt $stmt): string
     {
-        if ($property->isPrivate()) {
+        if ($stmt->isPrivate()) {
             return 'private';
         }
 
-        if ($property->isProtected()) {
+        if ($stmt->isProtected()) {
             return 'protected';
         }
 
         return 'public';
     }
 
     /**
-     * @return Property[][]
+     * @return Property[][]|ClassConst[][]
      */
     private function resolvePropertyByVisibilityByPosition(Class_ $class): array
     {
         $propertyByVisibilityByPosition = [];
         foreach ($class->stmts as $position => $classStmt) {
-            if (! $classStmt instanceof Property) {
+            if (! $classStmt instanceof Property && ! $classStmt instanceof ClassConst) {
```

Comment:
I see `OrderPropertyByComplexityRector` but here it orderes `ClassConst` as well? That's confusing.

It should be 2 separated rules or ClassConst skipped here.